src/python-common/ceph/deployment/service_spec.py

```diff
@@ -603,6 +603,10 @@ def validate(self) -> None:
         if self.service_type in self.REQUIRES_SERVICE_ID:
             if not self.service_id:
                 raise ServiceSpecValidationError('Cannot add Service: id required')
+            p = re.compile('^[a-zA-Z0-9_.-]+$')
+            if not p.match(self.service_id):
```

Comment:
but `p` is not reused after this line, would be better to just put
```py
if not re.match('^[a-zA-Z0-9_.-]+$', self.service_id):
```
to save the effort of `re.compile()` as the compiled regex object will be thrown away right after this line.